python/ray/data/datasource/parquet_base_datasource.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import (
+    Dict,
+    Any,
+    Callable,
+    TYPE_CHECKING,
+)
+
+if TYPE_CHECKING:
+    import pyarrow
+
+from ray.data.block import BlockAccessor
+from ray.data.datasource.file_based_datasource import (
+    FileBasedDatasource,
+    _resolve_kwargs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ParquetBaseDatasource(FileBasedDatasource):
```

Comment:
This class has only internal methods and also has only one subclass -- can it be folded into its subclass ParquetDatasource? My concern is it's a bit overskill for this case.

Reply:
I think that depends on what folding it into `ParquetDatasource` actually means but, if we mean getting rid of this class altogether, I don't think that will work moving forward. Some degree of refactoring may be possible, but we will need to separate this class in either this PR or the next PR since the upcoming Parquet bulk file reader API needs to use `ParquetBaseDatasource` directly. This API will not be able to use `ParquetDatasource` directly due to its `prepare_read` method override being the root cause of subsequent scalability/performance issues when consuming a large number of Parquet files cited in https://github.com/ray-project/ray/issues/22910. 

So we can delay separation of this class until the next PR, but I think it still needs to happen.

Reply:
Ok, thanks for explanation. Let's keep it then.